packages/reactstrap-validation-select/AvResourceSelect.js

```diff
@@ -97,36 +82,71 @@ class AvResourceSelect extends Component {
           throw new Error(`API returned an invalid response.`);
         }
         const getResult = this.props.getResult || this.props.resource.getResult;
+        let { hasMore } = this.props;
 
         const items =
           (typeof getResult === 'function'
             ? getResult.call(this.props.resource, resp.data)
             : resp.data[getResult]) || this.data;
 
+        hasMore = typeof hasMore === 'function' ? hasMore(resp.data) : hasMore;
+
         if (!Array.isArray(items)) {
           throw new TypeError(
             `Expected data to be an array but got \`${typeof items}\`. Use the \`getData\` prop to specify how to get the data from the API response.`
           );
         }
 
-        // eslint-disable-next-line promise/no-callback-in-promise
-        return callback(items);
+        return {
+          options: items,
+          hasMore,
+          additional: {
+            ...additional,
+            page: page + 1,
+          },
+        };
       })
       .catch(error => {
         throw error;
       });
-  }, this.props.delay);
+  };
 
   render() {
     const Tag = this.props.label ? AvSelectField : AvSelect;
+    const {
+      delay,
+      debounceTimeout,
```

Comment:
you can remove lines 125 and 126 by using default value in this destructing assignment.

```js
const {
  delay,
  debounceTimeout = delay,
  // ...
} = this.props;
```

Reply:
d8acb777f4937bc61785c9f249dd0d43b911a9fd

good catch 👍